cscs-checks/prgenv/src/alloc_speed.cpp

```diff
@@ -0,0 +1,27 @@
+#include <chrono>
+#include <iostream>
+#include <algorithm>
+
+double test_alloc(size_t n)
+{
+    auto t0 = std::chrono::system_clock::now();
+    /* time to allocate + fill */
+    char* ptr = (char*)std::malloc(n);
+    std::fill(ptr, ptr + n, 0);
+    auto t1 = std::chrono::system_clock::now();
+    /* time fo fill */
+    std::fill(ptr, ptr + n, 0);
+    auto t2 = std::chrono::system_clock::now();
+    t0 += static_cast<std::chrono::seconds>(ptr[0]);
```

Comment:
This is *VERY* cryptic. I understand you are doing it for fooling the compiler in order not to optimize out the `std::fill`, but you should explain. It is also quite error-prone. What if sb accidentally changes the array fill element to non-zero? Also, are you sure that none of the `std::fill()` is optimized out? I'd suggest declaring `ptr` as `volatile char *ptr` to avoid such optimizations from the compiler.

Reply:
`volatile` does not seem to work well with `unique_ptr`.